logicrunner/logicrunner.go

```diff
@@ -59,15 +59,17 @@ func (es *ExecutionState) Unlock() {
 
 // LogicRunner is a general interface of contract executor
 type LogicRunner struct {
-	Executors       [core.MachineTypesLastID]core.MachineLogicExecutor
+	MessageBus core.MessageBus `inject:""`
+	Ledger     core.Ledger     `inject:""`
+	Network    core.Network    `inject:""`
+
 	ArtifactManager core.ArtifactManager
```

Comment:
ArtifactManager isn't a component? Or just not yet?

Reply:
It's not a component. It's part of ledger.